framework/db/src/main/java/com/cloud/utils/db/Filter.java

```diff
@@ -77,7 +77,9 @@ public void addOrderBy(Class<?> clazz, String field, boolean ascending) {
 
         Column column = f.getAnnotation(Column.class);
         String name = column != null ? column.name() : field;
-
+        if (!name.equals("id")) {
+            name = String.format("%s, id", name);
```

Comment:
LGTM but I worry as this is used at so many places, @shwstppr - is there a more isolated way to fix the list API issue than this, i.e. what is causing the issue for the listServiceOfferings API response?

Reply:
@rhtyd as mentioned in the PR desc, it is sorting issue when entries in table have same value for the column (for which order by is called). In case of list* APIs where resources have `sort_key` in their corresponding tables.
Since by default value for `sort_key` is `0`, weirdly sorted results are returned when `LIMIT` offset is used. Change will add `id` as secondary sort criteria.

Only issue I see is the case when table doesn't have id column. Don't think that is the case with tables in ACS.

Reply:
What if the table has no `id` column @shwstppr the change is in `Filter` class and could have widespread side-effects. Can we check the service layer handler specific to listServiceOfferings and do something isolated/specific to that?

Reply:
@rhtyd changed the logic. No changes in Filter class now